### ISSUE TITLE
lsp-workspace-metadata "debug-sessions" is nil

### ISSUE THREAD

**Issue by belenot** [emacs-lsp/lsp-java#144]
With debugger I found that a function below every time return `nil`, meanwhile `lsp-session` isn't `nil`. Is it normal? I don't sure, because I suggest that it may be bug in my emacs config even before I tried lsp. 
(defun dap--get-sessions ()
  "Get sessions for WORKSPACE."
  (lsp-workspace-get-metadata "debug-sessions"))

In whole all works fine(hightlight, peek, flycheck...) except debugger. E.g when attaching it shows empty *host(port): out* buffer. 
Emacs `list-process` show running host:port, but linux's `sudo netstat -an` shows only listenable connection on host(port), while jvm is not attached. 
Sorry for such poor information, I am not strong enough in elisp :)

**Comment by yyoncho** [comment]
I am not 100% I understand what the issue is. Can you describe what are you trying to do(i. e. the steps that you are performing)? Are you saying that you are trying to attach to a process and it fails?

**Comment by belenot** [comment]
Um, it not fails. I didn't see errors, but dap-debug wasn't able to do anything. Worth mentioned that `dap-ui-sessions` shows empty list after the buffer *host(port)* appeared. And although I said that doesn't attach to jvm, tcp connection between emacs and jtdls exists

**Comment by yyoncho** [comment]
Can you provide a bit more details on what commands do you run? I assume that you are doing dap-debug, you select the Attach configuration and you enter the debug port but dap mode fails to attach to the process, is that what you are doing?

**Comment by belenot** [comment]
Yes, it is

**Comment by belenot** [comment]
So, I download package from [https://download.eclipse.org/jdtls/snapshots/?d](url) set `lsp-java-server-install-dir` to it and update server. When i try to attach to not existing port, it was throwing failed in `dap-ui-sessions`. I can't try it now, so tomorrow i'll say about result

**Comment by yyoncho** [comment]
@belenot you should not download the server manually but invoke lsp-java-update-server. If you download it manually the debug/test components wont be present.

**Comment by belenot** [comment]
@yyoncho , sorry, I couldn't reproduce the bug again( It may be caused by another version of java(when I first tried, it was openjdk 11). However, it didn't work last day with jdk8. 
All I can say, is that after I installed server manually, and made reference lsp to its path, all have work fine. Moreover, now, when I switch to first server installation, it's still working too) Maybe somedays I'll figure out what the issue is.
Thank you, it's a great job you've done already, I wish you good luck in further))
